ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/Update.scala

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v2
+
+import com.daml.lf.data.Time.Timestamp
+import com.daml.daml_lf_dev.DamlLf
+import com.daml.ledger.participant.state.v1.{
+  CommittedTransaction,
+  Configuration,
+  DivulgedContract,
+  ParticipantId,
+  TransactionId,
+  TransactionMeta
+}
+import com.daml.lf.transaction.BlindingInfo
+
+/** An update to the (abstract) participant state.
+  *
+  * [[Update]]'s are used in [[ReadService.stateUpdates]] to communicate
+  * changes to abstract participant state to consumers.
+  *
+  * We describe the possible updates in the comments of
+  * each of the case classes implementing [[Update]].
+  */
+sealed trait Update extends Product with Serializable {
+
+  /** Short human-readable one-line description summarizing the state updates content. */
+  def description: String
+
+  /** The record time at which the state change was committed. */
+  def recordTime: Timestamp
+}
+
+object Update {
+
+  /** Signal that the current [[Configuration]] has changed. */
+  final case class ConfigurationChanged(
+      recordTime: Timestamp,
+      submissionId: AdminSubmissionId,
+      participantId: ParticipantId,
+      newConfiguration: Configuration,
+  ) extends Update {
+    override def description: String =
+      s"Configuration change '$submissionId' from participant '$participantId' accepted with configuration: $newConfiguration"
+  }
```

Comment:
@andreaslochbihler-da : I'm thinking about how to implement the Ledger API server that handles both v1 and v2 ReadService backends, and it looks to me that a construction where a `v2.Update` is either a `v1.Update` or a `v2.TransactionAccepted` or a `v2.CommandRejected` would be a simple option with well-defined code-reuse properties. 

The constraint that a v2.ReadService does not generate v1.TransactionAccepted messages could then be enforced at runtime.

Once we have all ledgers updated to v2, we can then do the trivial refactoring of inlining the right v1 classes into v2 and enforce the structure at the type level.

What do you think?

Reply:
Let's wait until we have an overview of how many Updates remain the same and how much in the Ledger API server needs to change. I don't have a feeling that we'd keep a lot of code the same. If that's the case, then it might make sense to factor out a super-trait for the case classes that remain the same.

Reply:
Let's keep this idea in the comment.